### PR TITLE
allow currently installed plugins to be updated

### PR DESCRIPTION
**Pull request by ludicruz**
I noticed #11 after I had already gotten this to work.
I defaulted this to jenkins_plugin_update: False so there would be no backwards compatibility breaks.
Thank you

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -25,6 +25,10 @@ The location at which the `jenkins-cli.jar` jarfile will be kept. This is used f
       - sonar
       - ssh
 
+By default jenkins installs all your plugins but does not update any of the currently installed plugins. If you would like the currently installed plugins updated, set jenkins_plugin_update to true
```
Comment (geerlingguy) [inline README.md]: This description needs to go under the variable, and the whole addition needs to move under the block of text that follows it (my readmes go variable, then description).

File: tasks/plugins.yml [hunk 2]
```diff
@@ -27,3 +27,19 @@
     creates=/var/lib/jenkins/plugins/{{ item }}.jpi
   with_items: jenkins_plugins
   notify: restart jenkins
+
+- name: Get Plugins to update
+  # plugins which have an update show the newest available version in () at the right
+  # example:
+  # javadoc                           Javadoc Plugin                                   1.1 (1.3)
+  shell: java -jar {{ jenkins_jar_location }} -s http://{{ jenkins_hostname }}:8080/ list-plugins | grep "([0-9]\+\..*)" | awk '{print $1}'
```
Comment (geerlingguy) [inline tasks/plugins.yml]: To make this command a little more legible, can you use a folded scalar here?

```
  shell: >
    java -jar {{ jenkins_jar_location }} -s http://{{ jenkins_hostname }}:8080/ list-plugins |
    grep "([0-9]\+\..*)" |
    awk '{print $1}'
```

File: tasks/plugins.yml [hunk 3]
```diff
@@ -27,3 +27,19 @@
     creates=/var/lib/jenkins/plugins/{{ item }}.jpi
   with_items: jenkins_plugins
   notify: restart jenkins
+
+- name: Get Plugins to update
+  # plugins which have an update show the newest available version in () at the right
```
Comment (geerlingguy) [inline tasks/plugins.yml]: Comments should go above the `name` line, and I'd prefer a style that's just one proper sentence, wrap at 80 cols, e.g.:

```
  # Plugins which have an update show the newest available version in () at the
  # right, e.g. `javadoc    Javadoc Plugin    1.1 (1.3)`.
```

Can you update the comment to match?

File: tasks/plugins.yml [hunk 4]
```diff
@@ -27,3 +27,19 @@
...
+  # example:
+  # javadoc                           Javadoc Plugin                                   1.1 (1.3)
+  shell: java -jar {{ jenkins_jar_location }} -s http://{{ jenkins_hostname }}:8080/ list-plugins | grep "([0-9]\+\..*)" | awk '{print $1}'
+  register: pluginsToUpdate
```
Comment (geerlingguy) [inline tasks/plugins.yml]: Please change this var to match `snake_case` convention, e.g. `plugins_to_update`

**Comment by geerlingguy** [comment]
@ludicruz - This looks really helpful, and makes Jenkins usage even more automated (which is great!). Can you make the suggested changes in the code comments, and then I'll review once more and hopefully get it merged in?

**Comment by olo-dw** [comment]
I beg your pardon to interfere in this discussion, but this code seems to be outdated in regard of the actual master branch.
@ludicruz May I suggest to rebase and make the suggested changes?

**Pull request closed without merging**